### PR TITLE
Group list API

### LINKED ISSUE #98: Group list API
**Is your feature request related to a problem? Please describe.**
As a admin of the organisation i would like to view the list of groups available in the org.

image

**Describe the solution you'd like**
1. Create an api: `[get] /api/org/(org-id)/group/` which lists the groups available under that org.
1. It will return an array of groups objects which contain 
    1. group.id 
    1. group.name
    1. Number of members in that group (member_count)
(no need of role and actual permission info here as we can see from img)
1. only **staff** of that org can access this api.
1. make swagger docs for the same 
1. add tests for the same

### PR DESCRIPTION
**Pull request by KunalAnkur**
### Description

- Added the get request to get the data of all the groups.(only by staffs)

- updated the swagger doc.

- Implemented the tests cases.

**Type of Change:** 
- Code
- Testing

**Code/Quality Assurance Only** 

- New feature (non-breaking change which adds functionality pre-approved by mentors)

**How Has This Been Tested?**

- I am providing the test cases which ran successfully on my local system.

```
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
......................................
----------------------------------------------------------------------
Ran 38 tests in 6.565s

OK
Destroying test database for alias 'default'...
```

- When authorized user sent the get request to the server, then the server returns the following data.

Screenshot from 2020-05-27 10-40-42
 

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: org/views.py [hunk 1]
```diff
@@ -123,11 +123,15 @@ def Permission_Set_List(request,org_id):
         user = request.user,
         org = org
         )
+       
     except Member.DoesNotExist:
         return Response({"detail" : "You are not a member of this organisation"}, status.HTTP_400_BAD_REQUEST)
 
+    except Member.DoesNotExist:
+        return Response({"detail" : "You are not a member of this organisation"}, status.HTTP_400_BAD_REQUEST)
```
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
```

lines 127 & 128 are repeated

File: org/views.py [hunk 2]
```diff
@@ -137,4 +141,47 @@ def Permission_Set_List(request,org_id):
             }
             response_object.append(value)
         return Response(response_object, status.HTTP_200_OK)
-    return Response({"detail": "You are not authorised to view this."}, status.HTTP_403_FORBIDDEN)
\ No newline at end of file
+    return Response({"detail": "You are not authorised to view this."}, status.HTTP_403_FORBIDDEN) 
+
+@swagger_auto_schema(
+    operation_id="get_group",
```
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
    operation_id="get_groups_list",
```

File: org/views.py [hunk 3]
```diff
@@ -137,4 +141,47 @@ def Permission_Set_List(request,org_id):
...
+        '401': set_example(responses.user_unauthorized_401),
+        '404': set_example({"detail": "This organisation doesn't exist."}),
+        '400': set_example({"detail" : "You are not a member of this organisation"}),
+        '403': set_example({"detail": "You are not authorised to view this."}),
+    }
+)    
+@api_view(['get'])
+@permission_classes([IsAuthenticated])
+def GetGroup(request,org_id):
+    print(request.user)
```
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
```

Please remove the print statements used in the testing while making the PR

File: org/views.py [hunk 4]
```diff
@@ -137,4 +141,47 @@ def Permission_Set_List(request,org_id):
...
+    try:
+        member = Member.objects.get(
+        user = request.user,
+        org = org
+        )
+    except Member.DoesNotExist:
+        return Response({"detail" : "You are not a member of this organisation"}, status.HTTP_400_BAD_REQUEST)
+       
+    # if member.permission_set.perm_obj.permissions[Permissions.IS_STAFF]:
+    if True:
```
Comment (naveennvrgup) [inline org/views.py]: `if True:` seriously? :rofl: :rofl: :rofl:

File: tests/org/tests_get_group_api.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                    user = self.auth_user,
+                    org = self.org,
+                    group = admin_group,
+                    permission_set = admin_permission_set 
+                )
+            response = auth_client.get(get_group_api)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def tearDown(self):
+        self.auth_user.delete()
```
Comment (naveennvrgup) [inline tests/org/tests_get_group_api.py]: please call super class tearDown also

**Review by naveennvrgup — CHANGES_REQUESTED** [review]
Please rename test file as `tests_list_group_api.py`. I can see that all test functions are indented within the setUp(). So none of the tests are actually running.
image

Great effort :fire:

File: org/views.py (hunk 4 above)
Comment (KunalAnkur) [reply org/views.py]: sorry I forgot to remove that.. I was just checking that API 😅😅😅

**Comment by KunalAnkur** [comment]
@naveennvrgup  sir, I have updated my PR again. I am requesting you to review again. and next time I will be aware of those silly mistakes.

File: org/views.py [hunk 6]
```diff
@@ -137,4 +139,45 @@ def Permission_Set_List(request,org_id):
...
+        )
+    except Member.DoesNotExist:
+        return Response({"detail" : "You are not a member of this organisation"}, status.HTTP_400_BAD_REQUEST)
+       
+    if member.permission_set.perm_obj.permissions[Permissions.IS_STAFF]:    
+        group = Group.objects.all() 
+        response_object = []
+        for x in group:
+            memberLen = len(Member.objects.filter(group=x.id))
+            response_object.append({"groupId":x.id, "groupName": x.name, "groupMember":memberLen})
```
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
            response_object.append({"id":x.id, "name": x.name, "memberCount":memberLen})
```

Comment (naveennvrgup) [reply org/views.py]: we dont need the prefixing here

**Review by naveennvrgup — CHANGES_REQUESTED** [review]
you have named the file with `.py.py` (`tests/org/tests_list_group_api.py.py`). So the tests are not being detected. Run `./manage.py test -k GetGroupAPITestCase` to ensure that your tests are running :v: .

**Comment by sumit-badsara** [comment]
Resolve the conflicts in upcoming commit. @KunalAnkur

**Pull request merged**